trpcultivate_phenotypes/tests/src/Kernel/Services/Traits/ValidTraitTest.php

```diff
@@ -200,12 +222,141 @@ public function testTraitsServiceDatabaseExpectations() {
     $data_type = $this->chado_connection->query($sql, [
       ':c_id' => $trait_assets['unit'],
       ':t_id' => $this->terms['unit_type'],
-      ])->fetchObject();
+    ])->fetchObject();
 
     $this->assertNotNull($data_type, 'Failed to insert unit property - additional type.');
     $this->assertEquals($data_type->value, 'Quantitative', 'Unit property - additional type does not match expected value (Quantitative).');
   }
 
+  /**
+   * Test setTraitGenus() method for error cases.
+   */
+  public function testSetTraitGenus() {
+    // Test case for where the genus is not configured.
+    $exception_caught = FALSE;
+    $exception_message = 'NONE';
+    $exception_message = '';
+    try {
+      $this->service_traits->setTraitGenus('Test Genus');
+    }
+    catch (\Exception $e) {
+      $exception_caught = TRUE;
+      $exception_message = $e->getMessage();
+    }
+    $this->assertEquals(
+      'Exception: The genus Test Genus was not configured for' .
+      'use with Tripal Cultivate Phenotypes. To configure this genus, go to ' .
+      Url::fromRoute('trpcultivate_phenotypes.settings_ontology')->toString() .
+      ' and set the controlled vocabularies associated with this genus.',
+      $exception_caught,
```

Comment:
```suggestion
      $exception_message,
```
Since you're capturing values for $exception_caught and $exception_message, add an assert statement to verify respective values. Here's an excellent example from Carolyn https://github.com/TripalCultivate/TripalCultivate-Phenotypes/blob/b5b0450bb086e1318b5854394c2de1873bc40ef4/trpcultivate_phenotypes/tests/src/Kernel/Validators/ValidatorProjectGenusMatchProcessTest.php#L416